### PR TITLE
HTTP/GRPC API users configuration

### PR DESCRIPTION
**Pull request by pyakushin**
- [ ] secure storage of passwords

### CODE REVIEW

File: config-examples/users.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+users:
+  - name: default
+    password: default
```
Comment (agend) [inline config-examples/users.yaml:3]: never ever store password in plain string format.

File: config-examples/users.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+    perms: Read
```
Comment (agend) [inline config-examples/users.yaml:4]: It is not a good approach to creat internal user managment. Could create a lots of security issue. It's better to authorize via SAML, OpenID Connect.

File: bob/src/api/mod.rs [hunk 3]
```diff
@@ -1,2 +1,77 @@
...
+
+use bob_common::configs::{Access as AccessConfig, User as UserConfig};
+use rocket::{
+    http::Status,
+    request::{FromRequest, Outcome},
+    Request,
+};
+
+const BOB_API_USER_HEADER: &str = "x-bob-user-key";
+const BOB_API_PASSWORD_HEADER: &str = "x-bob-user-key";
```
Comment (agend) [inline bob/src/api/mod.rs:14]: same constant

File: config-examples/users.yaml (hunk 2 above)
Comment (agend) [reply config-examples/users.yaml:4]: bob is meant to be distributed. managing user in same way is not a nice.

**Comment by agend** [comment]
I think it's better to create gateway service

File: config-examples/users.yaml (hunk 1 above)
Comment (pyakushin) [reply config-examples/users.yaml:3]: Of course (: It's the first iteration to figure out the whole feature organization.

**Pull request closed without merging**